lib/modules/datasource/maven/util.ts

```diff
@@ -91,13 +93,78 @@ export async function downloadHttpProtocol(
       // istanbul ignore next
       logger.debug({ failedUrl }, 'Unsupported host');
     } else {
-      logger.info({ failedUrl, err }, 'Unknown error');
+      logger.info({ failedUrl, err }, 'Unknown HTTP download error');
     }
     return {};
   }
 }
 
-export async function checkHttpResource(
+function isS3CedentialsError(err: { name: string; message: string }): boolean {
+  return err.name === 'CredentialsProviderError';
+}
+
+function isS3RegionError(err: { name: string; message: string }): boolean {
+  return err.message === 'Region is missing';
+}
+
+function isS3NotFound(err: { name: string; message: string }): boolean {
+  return err.message === 'NotFound' || err.message === 'NoSuchKey';
+}
+
+function parseS3Url(rawUrl: url.URL | string): { Bucket: string; Key: string } {
```

Comment:
```suggestion
function parseS3Url(rawUrl: string): { Bucket: string; Key: string } {
```

`toString` is called before, so jo need to accept url object

Reply:
Done, also added `parseUrl` there.